### PR TITLE
#115 check and update hazard model

### PR DESCRIPTION
**Pull request by Mares2022**
The set up hazard was updated to create netcdf files with multiple bands pero return period map. The update integrates a small functionality to sorte the return period maps from the lower return period to the highest return period. Some check function as check_files functions and check_param_type  were removed or modified to avoid redundancy in the data checks. This included removing functions from validation.py. Functions to create list of parameters and check their size are still in the code. Data catalog is not used anymore in the set up hazard workflow so parameter catalog_name and its connections were removed.

### CODE REVIEW

**Comment by Mares2022** [comment]
Please add comments on how to improve the tests for the hazard components. I still miss to improve the tests. An addition point is that for one reason, when working with geotiffs the running time of the test is huge which makes me think that there is a wrong writing of the grid data. I already checked the data structure of the outputs but I require to check the spatial distribution of the maps (using qgis) to check if it is correct.

**Comment by Mares2022** [comment]
Test_hazard.py was erased in the current branch so we can erase it again in fiat_integrator.

**Review by frederique-hub — CHANGES_REQUESTED** [review]

File: hydromt_fiat/fiat.py [hunk 1]
```diff
@@ -363,58 +364,36 @@ def setup_hazard(
...
+                sfincs_model = SfincsModel(
+                    sfincs_root, mode="r", logger=self.logger
+                )
+                sfincs_model.read_results()
+                da = sfincs_model.results["zsmax"]
+                da = da.isel(timemax=0).drop("timemax")
+                # save sfincs map as GeoTIFF
+                # result_list = list(sfincs_model.results.keys())
+                # sfincs_model.write_raster("results.zsmax", compress="LZW")
+
```
[the remark this answers is not in the document]
Comment (Mares2022) [reply hydromt_fiat/fiat.py:389]: I removed all the sfinc components. This is a task that can be done through hydromt sfincs. I kept "da = da * unit_conversion_factor"  from here as a general check.

File: hydromt_fiat/workflows/hazard.py [hunk 2]
```diff
@@ -61,6 +61,7 @@ def check_parameters_type(
     dict
         Dictionary with the parameters and list of parameters used in setup_hazard.
     """
+    #TODO: remove this function
```
[the remark this answers is not in the document]
Comment (Mares2022) [reply hydromt_fiat/workflows/hazard.py:64]: The function was required partially. That is why par of it is still there, but a significant part was removed in hazard.py and validation.py. This function was modified to only produce a list of parameters, the check of data was already removed.

File: hydromt_fiat/fiat.py [hunk 3]
```diff
@@ -425,84 +404,86 @@ def setup_hazard(
...
-            # have the same name so give another name
-            if risk_output and da_map_fn.stem == "sfincs_map":
-                da_name = da_name + f"_{str(da_rp)}"
             map_name_lst.append(da_name)
+
+            da.attrs = {
+                "returnperiod": str(da_rp),
+                "type": da_type,
+                "name": da_name,
+                "analysis": "event",
```
[the remark this answers is not in the document]
Comment (Mares2022) [reply hydromt_fiat/fiat.py:420]: I will coordinate with Brandon. Attributes could be skipped in this dataset as the return period is written in the bands name of the dataset. I changed it as rp meanwhile.

**Comment by Mares2022** [comment]
This branch can already be merged, it addresses Dirk's comments. The only comment missing is this one: https://github.com/Deltares/hydromt_fiat/pull/124#discussion_r1360250835. We have to ensure that self.region exists to load raster data with the parameter geoms=self.region. At this moment, with the current test, it is working. I also need to improve the test, let me know what is missing. I removed the test for sfincs maps as they are not read anymore through hydromt fiat. The process of this sfincs.nc maps can be done through hydromt sfincs.

**Comment by frederique-hub** [comment]
Great! I will merge this branch into fiat_integrator to have everything there. We will discuss the remaining things tomorrow and add issues for those things if required. Good work!

**Pull request merged**